osbuild/util/bom/model.py

```diff
@@ -0,0 +1,207 @@
+"""Defines standard-agnostic data model for a BOM."""
+
+import urllib.parse
+import uuid
+from datetime import datetime
+from enum import Enum, auto
+from typing import Dict, List, Optional, Set
+
+
+class ChecksumAlgorithm(Enum):
+    SHA1 = auto()
+    SHA224 = auto()
+    SHA256 = auto()
+    SHA384 = auto()
+    SHA512 = auto()
+    MD5 = auto()
+
+
+class UniqueElement:
+    """Represents an element with a unique identifier."""
+
+    def __init__(self) -> None:
+        self.uuid: str = str(uuid.uuid4())
+
+    @property
+    def uuid(self) -> str:
+        """
+        Returns a stable UUID for the same RPM package as defined by the PURL.
+        """
+        return self._uuid
+
+    @uuid.setter
+    def uuid(self, value: str) -> None:
```

Comment:
I maybe rambling here but I wonder if the setter could just be removed? The class is called "UniqueElement" and it seems to me that providing a setter may compromise this uniqueness(?).

Maybe instead just make it an AbstractBaseClass that requires a uuid() property? Given that all child classes of UniqueElement override the two methods here that seems okay (but maybe there is more to it?)

Reply:
Thanks for your suggestion. I merged the `UniqueElement` class into `Package`, since it was not used by any other class. I also made `Package` an abstract class called `BasePackage` and made all of its methods abstract, because they were not really implemented.